drivers/SmartThings/zigbee-switch/src/zigbee-dual-metering-switch/init.lua

```diff
@@ -25,31 +32,91 @@ local function can_handle_zigbee_dual_metering_switch(opts, driver, device, ...)
   return false
 end
 
-local function endpoint_to_component(device, endpoint)
-  if endpoint == 2 then
-    return "switch1"
-  else
-    return "main"
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    for i = 1,1 do
+      local name = "AURORA Outlet 2"
+      local metadata = {
+        type = "EDGE_CHILD",
+        label = name,
+        profile = "switch-power-2",
+        parent_device_id = device.id,
+        parent_assigned_child_key = string.format("%02X", i),
+        vendor_provided_label = name,
+      }
+      driver:try_create_device(metadata)
+    end
   end
 end
 
-local function component_to_endpoint(device, component)
-  if component == "switch1" then
-    return 2
-  else
-    return 1
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZIGBEE then
+    device:set_find_child(find_child)
   end
 end
 
-local function map_components(self, device)
-  device:set_endpoint_to_component_fn(endpoint_to_component)
-  device:set_component_to_endpoint_fn(component_to_endpoint)
+local function switch_on_command_handler(driver, device, command)
+  device:send(OnOff.server.commands.On(device))
+  
+end
+
+local switch_off_command_handler = function(driver, device, command)
+  device:send(OnOff.server.commands.Off(device))
+end
+
+local function on_off_command_handler(driver, device, value, zb_rx)
+  local event
+  
+  if value == OnOff.server.commands.On.ID then
+    event = capabilities.switch.switch.on()
+    device:set_field(CURRENT_STATUS, "on")
+  elseif value == OnOff.server.commands.Off.ID then
+    event = capabilities.switch.switch.off()
+    device:set_field(CURRENT_STATUS, "off")
+  end
+  
+  if event ~= nil then
+    device:emit_event(event)
+  end
+end
+
+local function do_refresh(self, device)
+  device:send(OnOff.attributes.OnOff:read(device))
+  device:send(SimpleMetering.attributes.Divisor:read(device))
+  device:send(SimpleMetering.attributes.Multiplier:read(device))
 end
 
 local zigbee_dual_metering_switch = {
   NAME = "zigbee dual metering switch",
+  supported_capabilities = {
+    capabilities.switch,
+    capabilities.refresh,
+    capabilities.powerMeter
+  },
+  zigbee_handlers = {
+    cluster = {
+      [OnOff.ID] = {
+        [OnOff.server.commands.On.ID] = on_off_command_handler,
+        [OnOff.server.commands.Off.ID] = on_off_command_handler
+      }
+    }
+  },
+  capability_handlers = {
+    [capabilities.refresh.ID] = {
+      [capabilities.refresh.commands.refresh.NAME] = do_refresh
+    },
+    [capabilities.switch.ID] = {
+      [capabilities.switch.commands.on.NAME] = switch_on_command_handler,
+      [capabilities.switch.commands.off.NAME] = switch_off_command_handler
+    }
+  },
```

Comment:
it looks like you're overwriting default functions with copies of the default functions here. what am I missing?

Reply:
> it looks like you're overwriting default functions with copies of the default functions here. what am I missing?

no need to declare default functions at the beginning (I deleted it, thnx), but implementation in switch_on_command handler and switch_off_command handler are a bit different (in defaults we are sending commands to componets, here we don't have components, but devices)

Reply:
won't `command.component` just be "main", though?

`device:send` should be the same as `device:send_to_component("main"...`

Reply:
don't know why, but it works only if I write this method with this implementation:
`local function switch_on_command_handler(driver, device, command)
  device:send(OnOff.server.commands.On(device))
end`

Reply:
Did you get an error when you ran the tests locally? I'm trying to figure out why it wouldn't work.

Reply:
Unit test are ok when I run them loacally, but when I test the driver with device, it doesn't work correctly. Commands for parent device are not handled

Reply:
We should not override the defaults.  If there are issues, provide us logs so we can investigate why it's not functioning as expected.